module/VuFind/src/VuFind/Search/Solr/HierarchicalFacetHelper.php

```diff
@@ -199,6 +199,37 @@ public function formatDisplayText(
         return new TranslatableString($originalText, $displayText);
     }
 
+    /**
+     * Check if the given value is the deepest level in the facet list.
+     *
+     * Takes into account lists with multiple top levels.
+     *
+     * @param array  $facetList Facet list
+     * @param string $value     Facet value
+     *
+     * @return bool
+     */
+    public function isDeepestFacetLevel($facetList, $value)
+    {
+        $parts = explode('/', $value);
+        $level = array_shift($parts);
+        if (!is_numeric($level)) {
+            return false;
```

Comment:
It's just a safeguard to avoid trying to handle non-numeric value as numeric in case the index contains something unexpected, but I think you're right in that returning true makes more sense in that case. Fixed.

Reply:
Thanks!